hal/src/rtc/embassy/mod.rs

```diff
@@ -0,0 +1,96 @@
+use core::cell::RefCell;
+use core::task::Waker;
+
+use crate::pac::{interrupt, Interrupt, Osc32kctrl, Rtc, NVIC};
+use crate::rtc::modes::{
+    mode0::{Compare0, RtcMode0},
+    RtcMode,
+};
+use critical_section::{CriticalSection, Mutex};
+use embassy_time_driver::Driver;
+use embassy_time_queue_utils::Queue;
+
+embassy_time_driver::time_driver_impl!(static DRIVER: AtmelDriver = AtmelDriver{
+    queue: Mutex::new(RefCell::new(Queue::new())),
+
+});
+
+struct AtmelDriver {
+    queue: Mutex<RefCell<Queue>>,
+}
+
+impl AtmelDriver {
+    fn set_alarm(&self, _cs: &CriticalSection, at: u64, rtc: &Rtc) -> bool {
+        // Embassy uses u64::MAX as a "no upcoming interrupt" sentinel
+        let at = match u32::try_from(at) {
+            Ok(at) => at,
+            _ if at == u64::MAX => u32::MAX,
+            Err(_) => return false,
+        };
+
+        RtcMode0::set_compare(rtc, 0, at);
+        true
+    }
+}
+
+#[interrupt]
+fn RTC() {
+    critical_section::with(|cs| {
+        let rtc = unsafe { Rtc::steal() };
+        if RtcMode0::check_interrupt_flag::<Compare0>(&rtc) {
+            // Due to synchronization delay, the RTC may be slightly behind
+            // Assume the current time is the time the interrupt is set for
+            let now = RtcMode0::get_compare(&rtc, 0) as u64;
```

Comment:
@sjroe solved this by setting the prescaler to four, thus ensuring the resolution was coarse enough to not cause issues. I like this better, but am open to suggestions